batch_job/src/batch_job_mesos.c

```diff
@@ -0,0 +1,379 @@
+#include "batch_job.h"
+#include "batch_job_internal.h"
+#include "debug.h"
+#include "process.h"
+#include "macros.h"
+#include "stringtools.h"
+#include "path.h"
+#include "xxmalloc.h"
+#include "jx.h"
+#include "jx_print.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <signal.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <time.h>
+
+#define FILE_TASK_INFO "task_info"
+#define FILE_TASK_STATE "task_state"
+
+#define NUM_OF_TASKS 4096
+
+static int counter = 0;
+static int finished_tasks[NUM_OF_TASKS];
+static int num_finished_tasks = 0;
+static time_t old_time;
+static int is_first_time = 1;
+
+// mesos task struct
+typedef struct mesos_task{
+	int task_id;
+	char *task_cmd;
+	int num_input_files;
+	char **task_input_files;
+	int num_output_files;
+	char **task_output_files;
+} mesos_task;
+
+// get list of input files 
+char **build_str_lst_from_str(int *num_str, const char *str) 
```

Comment:
This looks a lot like `dttools/src/text_list.h` which loads an array of strings from a file.  Use that instead, just add a method to create a `text_list` from a string.